### PR TITLE
ByteUtils.arrayUnsignedComparator to replace Guava comparator

### PR DESCRIPTION
**Pull request by msgilligan**
Replace Guava UnsignedBytes.lexicographicComparator() with ByteUtils.arrayUnsignedComparator()

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/base/utils/ByteUtils.java [hunk 1]
```diff
@@ -356,4 +357,28 @@ public static boolean checkBitLE(byte[] data, int index) {
     public static void setBitLE(byte[] data, int index) {
         data[index >>> 3] |= bitMask[7 & index];
     }
+
+    /**
+     * Provides a byte array comparator.
+     * @return A comparator for byte[]
+     */
+    public static Comparator<byte[]> arrayUnsignedComparator() {
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/base/utils/ByteUtils.java:365]: I think this could use some tests?

Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/base/utils/ByteUtils.java:365]: Agreed.

Comment (schildbach) [reply core/src/main/java/org/bitcoinj/base/utils/ByteUtils.java:365]: (I do remember there were some pecularities about the ordering.)

Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/base/utils/ByteUtils.java:382]: Tests added.

**Comment by msgilligan** [comment]
@schildbach I added tests. If there's a test-case I missed, let me know and we can add it to the test vectors.

**Comment by schildbach** [comment]
Merged.

**Pull request closed without merging**